### PR TITLE
Enable Caesar fetching through a Workflow configuration key.

### PR DESCRIPTION
**Pull request by kieftrav**
Staging branch URL: https://pr-7148.pfe-preview.zooniverse.org

LInked Issue: https://github.com/zooniverse/front-end-monorepo/issues/6038
Linked FEM PR: https://github.com/zooniverse/front-end-monorepo/pull/6039

This PR does 2 things towards fetching Caesar reductions efficiently in FEM.
- First, it adds an Admin area toggle called "CaesarEnabled". This toggle, when enabled, shows part two.
- Second, it adds a small config section to the FEMLab workflow editing page - which is currently limited to a toggle - that enables this specific workflow to make requests from Caesar to fetch Reductions on Subject load in the Classifier
- The logic for using this workflow configuration in FEM is in a separate PR

# Required Manual Testing
- [ ] Does the Workflow Configuration area of FEMLab hide the Caesar Enabled section when the admin flag is false?
- [ ] Does the Workflow Configuration area of FEMLab show the Caesar Enabled section when the admin flag is true?

# Full Manual Testing
The full end-to-end connection between FEM & PFE for this PR can be reviewed as follows:

## Setup
- Open up the Admin section for the "[Annotate Or Not](https://local.zooniverse.org:3735/admin/project_status/kieftrav/annotate-or-not)" project
- Open up the Workflow Config for the "[Annotate Or Not](https://local.zooniverse.org:3735/lab/2002/workflows/3782)" project
- Open up the Classifier locally on the complementary branch for "[Annotate Or Not](https://localhost:8081/?project=kieftrav%2Fannotate-or-not&workflow=3782)"

## Testing
- With all 3 tabs open, you can toggle on and off the Caesar Enabled workflow setting in the Workflow Config page. If, for whatever reason it's not enabled (someone else testing), enable it in the Admin section of the first tab
- Go to the 3rd tab after enabling/disabling "Caesar Enabled" in the Workflow Config. Notice that when you refresh the Classifier page after each toggle that Caesar Reductions show up on the subject when enabled, and don't show up on the subject when disabled.

### CODE REVIEW

**Comment by kieftrav** [comment]
Admin Area Flag Screenshot & Workflow Config Screenshot attached..

admin-area-flag
workflow-config-flag

File: app/pages/lab-fem/workflow.jsx
Comment (lcjohnso) [inline app/pages/lab-fem/workflow.jsx:526]:
```suggestion
                    Caesar Data Fetching Enabled
```

Comment (lcjohnso) [reply app/pages/lab-fem/workflow.jsx:526]: Revised text makes clear this is related to data fetching, not related to other Caesar use cases (early retirement, etc).

File: app/pages/lab-fem/workflow.jsx
Comment (lcjohnso) [inline app/pages/lab-fem/workflow.jsx:532]:
```suggestion
                  <small className="form-help">Enabling Caesar data fetching allows the website to pull in subject reductions when the subject loads in the classifier.</small>
```

Comment (lcjohnso) [reply app/pages/lab-fem/workflow.jsx:532]: I'm suggesting that we keep the description short-and-sweet under the assumption the project teams will already have to be in touch with Zoo team in order to get the experimental feature activated via the admin panel.

File: app/pages/admin/project-status/experimental-features.jsx
Comment (lcjohnso) [inline app/pages/admin/project-status/experimental-features.jsx:8]:
```suggestion
  'caesarDataFetching'
```

Comment (lcjohnso) [reply app/pages/admin/project-status/experimental-features.jsx:8]: I'd opt for a more descriptive feature name such that comment is not necessary.

**Review by lcjohnso — CHANGES_REQUESTED** [review]
I recommend a number of text changes, mostly about what to call this feature. I'm open to discussion, but found it useful to layout my logic via specific code comments. Note: further changes would be necessary beyond my suggestions.

In addition to my specific suggestions: are we certain we want to hide this behind an experimental tool feature flag? The alternative would be to enable this config box for any relevant workflow in the FEMLab (and _only_ FEMLab): drawing task, transcription task, others?
- I'm leaning towards "hide behind experimental feature flag" because we don't want to support any/all teams who may inquire about the feature.
- Will the feature be auto-enabled for transcription task? And perhaps not able to be disabled in that case?

**Comment by kieftrav** [comment]
Thought I'd add a quick video that demonstrates the desired functionality between the Classifier and the Workflow Config.

https://github.com/user-attachments/assets/17af4b66-c220-43f6-ab94-5e5282220226

File: app/pages/lab-fem/workflow.jsx
Comment (lcjohnso) [inline app/pages/lab-fem/workflow.jsx:124]: Double checking: https://github.com/zooniverse/front-end-monorepo/pull/6039 assumes `enable_caesar_data_fetching` but here the configuration key is `enable_caesar_fetching` -- which are you planning to adopt?

**Comment by lcjohnso** [comment]
@kieftrav Checking in on this PR -- let me know when you're ready for a new review once keys / terminology has been corrected based on my comments and our Monday UMN+Cliff call discussion. As part of my review, I'll also perform the Panoptes DB check to identify cases where data fetching is currently enabled (and shouldn't be interrupted), and I can take on the task of manually adding the expected workflow configuration key for those so they work from the instant https://github.com/zooniverse/front-end-monorepo/pull/6039 merges.

And before this PR and https://github.com/zooniverse/front-end-monorepo/pull/6039 merges, let make sure we all understand the appropriate order of operations required, and we've double checked configuration setups, to prevent interruption of on-going projects.

**Comment by kieftrav** [comment]
@lcjohnso - sounds good. This PR has incorporated the changes you mentioned from above! Thing to clarify:

1) All transcription workflows are unaffected by this change. As per a much earlier discussion the idea was to leave them unaffected
2) All FreehandLine workflows are the only other workflows I know of successfully using Caesar Reductions.

Looking forward to next steps!

**Comment by lcjohnso** [comment]
@kieftrav -- Is this branch up-to-date? I'm still seeing the original "Caesar Enabled" nomenclature as opposed to the "Caesar Data Fetching" naming convention. Similarly, my open question regarding the expected workflow.configuration key still stands.

**Comment by kieftrav** [comment]
@lcjohnso - my mistake on not getting it pushed up from local. It's all in there now and ready for review.

File: app/pages/lab-fem/workflow.jsx
Comment (kieftrav) [reply app/pages/lab-fem/workflow.jsx]: I had to fix an errant file getting added to the commit that shouldn't been in there and didn't push up the appropriate change. This should be fixed and matching FEM.

**Review by lcjohnso — APPROVED** [review]
Looks good now! It would be good to have another dev take a look, but A-OK from my perspective.

**Comment by lcjohnso** [comment]
@kieftrav I still need to do my workflow checks and enabling of the `enable_caesar_data_fetching` configuration key for relevant freehand drawing workflows before this merges along with https://github.com/zooniverse/front-end-monorepo/pull/6039, so lets be sure to touch base before moving forward.

**Review by shaunanoordin — APPROVED** [review]
## PR Review

This PR allows the (FEM Lab) Workflow to explicitly mark workflows to use the new(ly generalised) "Caesar Data Fetching" feature on FEM.

- Basically, we've got a new control that sets the new config value, `workflow.configuration.enable_caesar_data_fetching`, to either `true` or `false`. Pretty straightforward.
- The "enable Caesar Data Fetching" option ONLY appears the FEM Lab Workflow Editor.
  - ...and only when the 'caesarDataFetching' experimental flag is enabled.
  - Note: as with all workflow configuration controls, un-checking the checkbox will set it to `workflow.configuration.enable_caesar_data_fetching = false`, it won't erase the item from `workflow.configuration`

See also zooniverse/front-end-monorepo#6199 for the ADR (documentation, context) and zooniverse/front-end-monorepo#6039 for the corresponding changes to the FEM Classifier that makes use of the workflow configuration.

### Status

Changes look good, as this is a relatively simple new workflow configuration toggle, hidden behind a new experimental tool flag. I see no conflicts with what's happening on the corresponding FEM PR.

LGTM 👍

**Pull request merged**
